### PR TITLE
Add support for parsing JSX

### PR DESCRIPTION
**Pull request by arv**
This adds support for Facebook's JSX syntax.

https://facebook.github.io/jsx/

The only thing that is not implemented is JSXSpreadAttribute.

The option to turn this on is called `jsx`. You can provide your own
function to call by setting this option to a string. `--jsx=my.func`.
If not set it defaults to `React.createElement`.

### CODE REVIEW

**Comment by arv** [comment]
@johnjbarton I'm back!

File: src/codegeneration/JsxTransformer.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+/**
+ * Desugars JSX expressions.
+ *
+ *   <p a="b" c="d">e{f}g</p>
+ *
+ * to:
+ *
+ *   Rect.createElement('p', {a: 'b', c: 'd'}, 'e', f, 'g')
+ *
+ * The function to use is be default React.createElement but it is configurable.
```
Comment (johnjbarton) [inline src/codegeneration/JsxTransformer.js]: I don't understand the comment.

Comment (johnjbarton) [inline src/codegeneration/JsxTransformer.js]: I guess you meant: 
The constructor emitted is configurable; it defaults to Rect.createElement().

**Comment by johnjbarton** [comment]
LGTM

File: src/codegeneration/JsxTransformer.js (hunk 1 above)
Comment (arv) [inline src/codegeneration/JsxTransformer.js]: Reworded it a bit.

**Pull request merged**